File: 00_arrays_and_hashing/01_valid_anagram/valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        track = dict()
        for letter in "abcdefghijklmnopqrstuvwxyz":
            track[letter] = 0
        for letter in s:
            track[letter] += 1
        for letter in t:
            track[letter] -= 1
        for letter_count in track:
            if track[letter_count] != 0:
                return False
        return True


def test():
    solution = Solution()
    solution.isAnagram("rat", "car")


if __name__ == "__main__":
    test()

File: arrays & hashing/01_valid_anagram/valid_anagram.cpp
#include <string>

class Solution
{
  public:
    bool isAnagram(std::string s, std::string t)
    {
        if (s.size() != t.size())        
            return false;
        unsigned int track[26];
        std::string::const_iterator iter_s = s.begin();
        std::string::const_iterator iter_t = t.begin();
        while (iter_s != s.end())
        {
            ++(track[*iter_s - 'a']);
            --(track[*iter_t - 'a']);
            ++iter_s;
            ++iter_t;
        }
        for (char i = 0; i < 26; ++i)
            if (track[i] != 0)
                return false;
        return true;
    }
};

#include <iostream>

int main()
{
    Solution solution;
    std::string s("rat");
    std::string t("car");
    std::cout << "Valid anagram: " << solution.isAnagram(s, t) << "\n";
}
